integration-tests/test_hub.py

```diff
@@ -37,13 +38,41 @@ async def test_user_code_execute():
     assert 0 == await (await asyncio.create_subprocess_exec(*TLJH_CONFIG_PATH, 'reload')).wait()
 
     async with User(username, hub_url, partial(login_dummy, password='')) as u:
-            await u.login()
-            await u.ensure_server_simulate()
-            await u.start_kernel()
-            await u.assert_code_output("5 * 4", "20", 5, 5)
+        await u.login()
+        await u.ensure_server_simulate()
+        await u.start_kernel()
+        await u.assert_code_output("5 * 4", "20", 5, 5)
 
-            # Assert that the user exists
-            assert pwd.getpwnam(f'jupyter-{username}') is not None
+        # Assert that the user exists
+        assert pwd.getpwnam(f'jupyter-{username}') is not None
+
+
+@pytest.mark.asyncio
+async def test_user_code_execute_with_custom_base_url():
+    """
+    User logs in, starts a server with a custom base_url & executes code
+    """
+    # This *must* be localhost, not an IP
+    # aiohttp throws away cookies if we are connecting to an IP!
+    hub_url = 'http://localhost'
```

Comment:
The hub url should use the `base_url` we're setting for this test:
```python
hub_url = f"http://localhost{base_url}"
```

Reply:
My bad, now it's added.